collections/ansible_collections/cloudkit/service/roles/cleanup_stale_network_resources/tasks/cleanup.yaml

```diff
@@ -0,0 +1,245 @@
+---
+- name: Get all existing ClusterOrders from specified namespace
+  kubernetes.core.k8s_info:
+    api_version: cloudkit.openshift.io/v1alpha1
+    kind: ClusterOrder
+    namespace: "{{ cleanup_stale_network_resources_namespace }}"
+  register: cluster_orders
+  failed_when: false
+
+- name: Initialize existing cluster names list
+  ansible.builtin.set_fact:
+    existing_cluster_names: []
+
+- name: Extract cluster names from ClusterOrders
+  ansible.builtin.set_fact:
+    existing_cluster_names: "{{ existing_cluster_names + ['clusterorder_' + item.metadata.name] }}"
+  loop: "{{ cluster_orders.resources }}"
+  when: cluster_orders.resources is defined
+
+- name: Display existing cluster names
+  ansible.builtin.debug:
+    msg: |
+      Found existing ClusterOrders: {{ existing_cluster_names }}
+
+# Networks
+- name: Get all networks with purpose_o-sac and instance tags
+  ansible.builtin.command: >-
+    openstack network list
+    --tag {{ cleanup_stale_network_resources_osac_tag }}
+    --tag instance_{{ cleanup_stale_network_resources_namespace }}
+    -f json
```

Comment:
We don't want to call `openstack` directly in any of the cloudkit playbooks (if you check, you'll see we explicitly avoid doing so). Instead, you'll want to create a new `massopencloud.esi.l2` task that does this. I'd suggest having that task take in a `property` parameter, similar to what was used when creating the network, and then having the task translate that into a tag.